herddb-core/src/main/java/herddb/cluster/BookkeeperCommitLog.java

```diff
@@ -345,19 +402,20 @@ private void handleBookKeeperAsyncFailure(Throwable cause, LogEntry edit) {
                     + tableSpaceDescription(), cause);
         } else if (cause instanceof BKException.BKLedgerFencedException) {
             LOGGER.log(Level.SEVERE, "this server was fenced for tablespace " + tableSpaceDescription() + " !", cause);
-            close();
```

Comment:
It doesn't need close anymore here?

Reply:
now this code is executed during the write callback.

"closing" a WriteHandle is a metadata operation, and it errors out every other pending reads.

unfortunately it is named "close" but it should be named "seal", it is not about releasing resources but about writing to ZK current state of the ledger (LastAddConfirmed)

This is not supposed to be done inside the write callback.
We are marking the writer as "failed" any other "recovery" will be performed at a later stage
when the leader starts again and it opens the ledger for "recovery"